fe/fe-core/src/main/java/org/apache/doris/mysql/MysqlSslContext.java

```diff
@@ -0,0 +1,277 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.mysql;
+
+import org.apache.doris.common.Config;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLEngineResult;
+import javax.net.ssl.SSLEngineResult.HandshakeStatus;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.TrustManagerFactory;
+
+public class MysqlSslContext {
+
+    private static final Logger LOG = LogManager.getLogger(MysqlSslContext.class);
+    private SSLEngine sslEngine;
+    private SSLContext sslContext;
+    private String protocol;
+    private ByteBuffer serverAppData;
+    private static final String keyStoreFile = Config.mysql_ssl_default_certificate;
+    private static final String trustStoreFile = Config.mysql_ssl_default_certificate;
+    private static final String certificatePassword = Config.mysql_ssl_default_certificate_password;
+    private ByteBuffer serverNetData;
+    private ByteBuffer clientAppData;
+    private ByteBuffer clientNetData;
+
+    public MysqlSslContext(String protocol) {
+        this.protocol = protocol;
+    }
+
+    public void init() {
+        initSslContext();
+        initSslEngine();
+    }
+
+    private void initSslContext() {
+        try {
+            KeyStore ks = KeyStore.getInstance("PKCS12");
+            KeyStore ts = KeyStore.getInstance("PKCS12");
+
+            char[] password = certificatePassword.toCharArray();
+
+            ks.load(Files.newInputStream(Paths.get(keyStoreFile)), password);
+            ts.load(Files.newInputStream(Paths.get(trustStoreFile)), password);
+
+            KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(ks, password);
+
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(ts);
+            sslContext = SSLContext.getInstance(protocol);
+            sslContext.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException | IOException
+                 | CertificateException | UnrecoverableKeyException e) {
+            LOG.error("Failed to initialize SSL because", e);
```

Comment:
Do we need to exit the entire FE process if something is wrong (e.g. no such file)?
If not, can FE run without SSL support properly when it fails to init ssl context here?